README.md

```diff
@@ -1,111 +1,282 @@
+[![GoDoc](https://godoc.org/github.com/google/cel-go?status.svg)][6]
 
-This repo contains the Go toolchain for the Common Expression Language (CEL).
-CEL is a non-Turing complete language designed to be portable and fast. It is
-best suited to  applications where sandboxing a full-fledged language like
-JavaScript or Lua would be too resource intensive, but side-effect free dynamic
-computations are strongly desired.
+The Common Expression Language (CEL) is a non-Turing complete language designed
+for simplicity, speed, safety, and portability. CEL's C-like [syntax][1] looks
+nearly identical to equivalent expressions in C++, Go, Java, and TypeScript.
 
-## Want to learn more?
-
-* See the [CEL Spec][1] for more details about the language.
-* Ask for support on the [CEL Go Discuss][2] Google group.
-
-## Want to integrate CEL?
-
-* See [GoDoc][6] to learn how to integrate CEL into services written in Go.
-  [![GoDoc](https://godoc.org/github.com/google/cel-go?status.svg)][6]
-* See the [CEL C++][3] toolchain (under development) for information about how
-  to integrate CEL evaluation into other environments.
+```java
+// Check whether a resource name starts with a group name.
+resource.name.startsWith("/groups/" + auth.claims.group)
+```
 
-## Want to contribute?
+```go
+// Determine whether the request is in the permitted time window.
+request.time - resource.age < duration("24h")
+```
 
-* See [CONTRIBUTING.md](./CONTRIBUTING.md) to get started.
-* Use [GitHub Issues][4] to request features or report bugs.
+```typescript
+// Check whether all resource names in a list match a given filter.
+auth.claims.email_verified && resources.all(r, r.startsWith(auth.claims.email))
+```
 
-## How does CEL work?
+A CEL "program" is a single expression. The examples have been tagged as
+`java`, `go`, and `typescript` within the markdown to showcase the commonality
+of the syntax.
 
-Write an expression in the CEL syntax, then parse, check, and interpret.
+CEL is ideal for lightweight expression evaluation when a fully sandboxed
+scripting language is too resource intensive.
 
-| Step      | Description                                                    |
-|-----------|----------------------------------------------------------------|
-| Parse     | Parses a string expression to a protocol buffer representation.|
-| Check     | Type-checks a parsed expression against a given environment.   |
-| Interpret | Evaluates parsed expressions against a set of inputs.          |      |
+## Overview
 
-Type-checking an expression in an optional, but strongly encouraged step that
-can be used to reject some expressions as semantically invalid using static
-analysis. Additionally, the type-check produces some additional metadata
-related to function overload resolution and object field selection which may
-be used by the interpret step to speed up execution.
+Determine the attribute names and functions you want to provide to the CEL
```

Comment:
"Variable names" instead of "attribute names".  Attributes refer to properties of an underlying entity.